Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -46,25 +46,25 @@ void gdsp_ifx_init()
 		g_dsp.ifx_regs[i] = 0;
 	}
 
-	g_dsp.mbox[0] = 0;
-	g_dsp.mbox[1] = 0;
+	g_dsp.mbox[0].store(0);
+	g_dsp.mbox[1].store(0);
 }
 
 u32 gdsp_mbox_peek(u8 mbx)
 {
-	return Common::AtomicLoad(g_dsp.mbox[mbx]);
+	return g_dsp.mbox[mbx].load();
 }
 
 void gdsp_mbox_write_h(u8 mbx, u16 val)
 {
-	const u32 new_value = (Common::AtomicLoadAcquire(g_dsp.mbox[mbx]) & 0xffff) | (val << 16);
-	Common::AtomicStoreRelease(g_dsp.mbox[mbx], new_value & ~0x80000000);
+	const u32 new_value = (g_dsp.mbox[mbx].load() & 0xffff) | (val << 16);
```

Comment:
If you want to do it right, this should really be a compare_and_exchange. Otherwise using atomics doesn't provide much value except feeling good.